pkg/services/provisioning/datasources/config_reader.go

```diff
@@ -83,7 +83,7 @@ func (cr *configReader) parseDatasourceConfig(path string, file os.FileInfo) (*D
 }
 
 func validateDefaultUniqueness(datasources []*DatasourcesAsConfig) error {
```

Comment:
Not a big fan of us returning an error here when we could return information about what went wrong and log a user-friendly message instead.

Not really related to your changes though.

Reply:
Agree and agree

Reply:
Also, we still want an error in the next step so that we can break out of provisioning.